std/experimental/allocator/package.d

```diff
@@ -443,6 +449,21 @@ auto make(T, Allocator, A...)(auto ref Allocator alloc, auto ref A args)
         return emplace!T(m, args);
     else return emplace(cast(T*) m.ptr, args);
 }
+/// ditto
+auto make(T, Allocator, P, A...)(auto ref Allocator alloc, auto ref P outer, auto ref A args)
+    if (is(T == class) && isInnerClass!T && is(P : AliasSeq!(__traits(parent, T))[0]))
+{
+    import std.algorithm.comparison : max;
+    import std.conv : emplace;
+    auto m = alloc.allocate(max(stateSize!T, 1));
+    if (!m.ptr) return null;
+    scope(failure) alloc.deallocate(m);
```

Comment:
could you add a test for this case?
You just need a Postblit constructor that throws, e.g.:

https://github.com/dlang/phobos/blob/master/std/experimental/allocator/package.d#L1056
https://github.com/dlang/phobos/blob/master/std/experimental/allocator/package.d#L787